### PR TITLE
Set dbt asset check severity at runtime

### PR DESCRIPTION
**Pull request by johannkm**
We were reading from the manifest. Instead read from the event so we can support `error_if/warn_if`

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #19353** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19353?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19353?utm_source=stack-comment).

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_asset_checks/seeds/raw_customers.csv [hunk 1]
```diff
@@ -98,4 +98,4 @@ id,first_name,last_name
 97,Shirley,D.
 98,Nicole,M.
 99,Mary,G.
-100,Jean,M.
+100,Jean,M.
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_asset_checks/seeds/raw_customers.csv:101]: revert [1]

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_asset_checks/seeds/raw_payments.csv [hunk 2]
```diff
@@ -111,4 +111,4 @@ id,order_id,payment_method,amount
 110,96,gift_card,1700
 111,97,bank_transfer,1400
 112,98,bank_transfer,1000
-113,99,credit_card,2400
+113,99,credit_card,2400
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_asset_checks/seeds/raw_payments.csv:114]: [1]

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_asset_checks/models/schema.yml [hunk 3]
```diff
@@ -86,3 +86,20 @@ models:
...
         tests:
           - not_null
+
+  - name: fail_tests_model
+    columns:
+      - name: id
+        tests:
+          - unique:
+              config:
+                error_if: ">10"
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_asset_checks/models/schema.yml:96]: Just to be explicit that we want to skip this error condition?


```suggestion
                severity: error # This should never error, and instead we warn at run time
                error_if: "=-1" 
```

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_asset_checks/models/schema.yml [hunk 4]
```diff
@@ -86,3 +86,20 @@ models:
...
+                warn_if: ">0"
+
+      - name: first_name
+        tests:
+          - accepted_values:
+              values: ["foo", "bar", "baz"]
+              config:
+                error_if: ">3"
+                warn_if: ">0"
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_asset_checks/models/schema.yml:105]:
```suggestion
              config:
                severity: error
                error_if: "!=0"
```

**Review by rexledesma — COMMENTED** [review]
remove unnecessary whitespace changes + let's remove the unnecessary config in tests

**Review by rexledesma — APPROVED** [review]

**Pull request merged**